core/src/helpers.rs

```diff
@@ -169,16 +172,281 @@ pub fn starknet_block_to_eth_block(block: MaybePendingStarknetBlock) -> RichBloc
+                let parent_hash = H256::from_slice(&block_with_txs.parent_hash.to_bytes_be());
+                let sequencer =
+                    H160::from_slice(&block_with_txs.sequencer_address.to_bytes_be()[12..32]);
+                let state_root = H256::from_slice(&block_with_txs.new_root.to_bytes_be());
+                let number = U256::from(block_with_txs.block_number);
+                let timestamp = U256::from(block_with_txs.timestamp);
+                let transactions = BlockTransactions::Full(
+                    block_with_txs
+                        .transactions
+                        .into_iter()
+                        .map(from_starknet_transaction_ethers_transaction)
+                        .collect(),
+                );
+                let header = Header {
+                    hash: Some(hash),
+                    parent_hash,
+                    uncles_hash: parent_hash,
+                    author: sequencer,
+                    miner: sequencer,
+                    state_root,
+                    // BlockWithTxHashes doesn't have a transactions root
+                    transactions_root: H256::zero(),
+                    // BlockWithTxHashes doesn't have a receipts root
+                    receipts_root: H256::zero(),
+                    number: Some(number),
+                    gas_used,
+                    gas_limit,
+                    extra_data,
+                    logs_bloom,
+                    timestamp,
+                    difficulty,
+                    nonce,
+                    size,
+                };
+                let block = Block {
+                    header,
+                    total_difficulty,
+                    uncles: vec![],
+                    transactions,
+                    base_fee_per_gas: None,
+                    size,
+                };
+                Rich::<Block> {
+                    inner: block,
+                    extra_info: BTreeMap::default(),
+                }
             }
         },
     }
 }
 
+fn from_starknet_transaction_ethers_transaction(tx: StarknetTransaction) -> EtherTransaction {
+    let mut ether_tx = EtherTransaction::default();
+
+    match tx {
+        StarknetTransaction::Invoke(invoke_tx) => {
+            match invoke_tx {
+                InvokeTransaction::V0(v0) => {
+                    // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+                    ether_tx.hash = H256::from_slice(&v0.transaction_hash.to_bytes_be());
+                    ether_tx.nonce = field_element_to_u256(v0.nonce);
+                    ether_tx.from = starknet_address_to_ethereum_address(v0.contract_address);
+                    // Define gas_price data
+                    ether_tx.gas_price = None;
+                    // Extracting the signature
+                    ether_tx.r = field_element_to_u256(v0.signature[0]);
```

Comment:
to avoid possible panic, let's use `v0.signature.get(0)`, it'll return an option, from there, we can choose how to handle the `None` case properly

Reply:
wdyt, we can choose to return an custom error, a default 0 value, etc.

Reply:
Yeap, makes sense